### DISCUSSION TITLE
Match where all related documents have a valid value

### DISCUSSION THREAD

**Discussion by MorenoMdz** [prisma/prisma#20279] [Q&A]
Hey team, I have one part of our code that we use Prisma Raw to query this:
```js
        .map(
          (val) =>
            `EXISTS (
        SELECT 1
        FROM "profiles"
        WHERE "user" = "userId"."id" 
          AND "profiles"."profileField" = '${val}' 
          AND "profiles"."status" IN ('REPORTED', 'VERIFIED')
      )`,
        )
        .join(' AND ')
```
basically we need to match al users that have profile records for ALL the profileField types in the request.

Can we do this with Prisma natively? Using every will only match on the same record, need to have it at a relation level, since these are unique by profileField maybe I could use a profiles count = N of val?

**Comment by ludralph** [comment]
Hi @MorenoMdz 👋 

Can you please share the schema file?

**Reply by nurul3101** [reply]
Hey @MorenoMdz 👋 

Just checking to see if you still need any assistance. Do let us know.

**Reply by MorenoMdz** [reply]
Sorry getting used to use GH as a forum haha

```js
model Worker {
  id                         String                       @id @default(uuid())
  createdAt                  DateTime                     @default(now())
  updatedAt                  DateTime                     @updatedAt
 ...
  workerProfile              WorkerProfile[]
  ...
...
  @@map("workers")
}


model WorkerProfile {
  id           String   @id @default(uuid())
  createdAt    DateTime @default(now())
  updatedAt    DateTime @updatedAt
  status       String
  profileField String
  category     String
  value        Boolean?
  worker       Worker   @relation(fields: [workerId], references: [id], onDelete: Cascade)
  workerId     String

  @@unique([workerId, profileField])
  @@map("worker_profiles")
}

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by MorenoMdz** [reply]
Hey @ludralph I have responded to the thread, but I got no updates from the Prisma team, I think this one should be open, and would be great to have some eyes on this request.
